Automattic/woocommerce-payments
# Add workflow "Build live branch for PR"

Fixes #5624

#### Changes proposed in this Pull Request

Add a new GitHub workflow, besides building the file, and notifying the Jetpack Beta builder (betadownload.jetpack.me), it would do something more for that purpose: 

- Play around the version to have the dev version, and modify the built files.
- Tweak the file folder of artifacts in accordance with Jetpack Beta Builder. See p1678811397206069/1678259021.463009-slack-CBG1CP4EN
- Add and update comment to indicate where to download the zip file. See the `botwoo` comment below. This would be useful to download the build file too. I think it might be helpful for people outside A8C.

A note on this comment: 

> Building a zip file already exists: https://github.com/Automattic/woocommerce-payments/blob/develop/.github/workflows/build-zip-and-run-smoke-tests.yml
> 
> Source: https://github.com/Automattic/woocommerce-payments/issues/5624#issuecomment-1464028911

I am not using this workflow because:

1. the artifact retention day for live branch needs to be only 1 day.
2. Jetpack Beta Builder, i.e. betadownload.jetpack.me, has a specific requirement for the artifact. 









#### Testing instructions

- Check the run for the new action in this PR.
- Check the comment from `botwoo`.
- Confirm: https://betadownload.jetpack.me/woocommerce-payments-branches.json showing the new file for this PR. (Usually, it's almost instant, but sometimes I guess you might wait a bit). 

It would be similar to other files such as https://betadownload.jetpack.me/woocommerce-branches.json, which is eventually used to populate the options for WCPay in Jurassic Ninja.





*

-------------------


**Post merge**
